File: ClinicSystem/src/com/example/clinicsystem/controllers/C_Appointment.java
package com.example.clinicsystem.controllers;

import com.example.clinicsystem.models.classes.M_Appointment;
import com.example.clinicsystem.models.classes.M_Patient;
import com.example.clinicsystem.models.enums.DentistryDepartment;

import javax.swing.*;
import java.util.ArrayList;
import java.util.List;

public class C_Appointment {
    public List request(String requestType, List list) {
        M_Appointment appointment = new M_Appointment();

        switch (requestType) {
            case "C":
                appointment.setPatientID(Integer.parseInt(list.get(0).toString()));
                appointment.setPatientName(list.get(1).toString());
                appointment.setDentistID(Integer.parseInt(list.get(2).toString()));
                appointment.setDentistName(list.get(3).toString());
                appointment.setStatus("Pending");
                appointment.setAppointmentType(DentistryDepartment.getConstant(list.get(4).toString()));
                appointment.setDate(list.get(5).toString());
                appointment.setByPhone((boolean) list.get(6));
                appointment.setStartTime(list.get(7).toString());
                appointment.setEndTime("00:00");
                appointment.setID(appointment.addSchedule());
                appointment.addAppointment();

                return null;
            case "R":
                if (list == null) {
                    return appointment.getAllAppointments("");
                } else {
                    return appointment.getAllAppointments(list.get(0).toString());
                }

            case "U":

            case "D":

        }

        return null;
    }

    public boolean isValidAppointmentData(int PID) {
        C_Patient patientController = new C_Patient();
        List<String> data = new ArrayList<>();
        String condition = " WHERE Patient_ID = " + PID;
        data.add(condition);
        List<M_Patient> patients = patientController.request("R", data);

        if (patients.size() > 0) {
            return true;
        }

        return false;
    }
}

File: clinic/C_Appointment.php
<?php
include_once "M_Appointment.php";
include_once "C_Patient.php";

class C_Appointment {
    public function request($requestType, $data) {
        $appointment = new M_Appointment(1, "", 1, 1, FALSE, "", "", "", "", "", 1);
		
        switch ($requestType) {
            case "C":
                $appointment->setPatientID($data[0]);
                $appointment->setPatientName($data[1]);
                $appointment->setDentistID($data[2]);
                $appointment->setDentistName($data[3]);
                $appointment->setStatus("Pending");
                //$appointment->setAppointmentType(DentistryDepartment.getConstant($data.get(4).toString()));
                $appointment->setsDate($data[5]);
                $appointment->setByPhone($data[6]);
                $appointment->setStartTime($data[7]);
                $appointment->setEndTime("00:00");
                $appointment->setID($appointment->addSchedule());
                $appointment->addAppointment();

                return null;
            case "R":
				if ($data == null) {
                    return $appointment->getAllAppointments("");
                } else {
                    return $appointment->getAllAppointments($data[0]);
                }
            case "U":

            case "D":
			
			default:
				return null;
        }
    }

    public function isValidAppointmentData($PID) {
        $patientController = new C_Patient();
        $data = array();
        $condition = " WHERE Patient_ID = " . $PID;
		array_push($data, $condition);
        $patients = $patientController->request("R", $data);

        if (sizeof($patients) > 0) {
            return TRUE;
        }

        return FALSE;
    }
}
?>